core/src/main/scala/com/github/johnynek/paiges/Doc.scala

```diff
@@ -110,6 +112,18 @@ sealed abstract class Doc extends Serializable {
   def writeTo(maxLine: Int, pw: PrintWriter): Unit =
     Doc.write(this, maxLine, pw)
 
+  override lazy val hashCode: Int = {
```

Comment:
I wonder if while we are at it making a `Long` hash is worth it, and then `def hashCode` just gets the low 32 bits. For algorithms like HLL or CMS, we would like a longer hash. Can punt to later. I don't see us doing sketches of Docs real soon.